### DISCUSSION TITLE
[Arcropolis 3.5.0] [Character Select Screen Crash + Software Close] on 2+ Player/CPU/amiibo Smash

### DISCUSSION THREAD

**Discussion by arcenedon** [Raytwo/ARCropolis#370] [Issues] [answered]
System Firmware Version
14.1.1

Atmosphere Version
1.3.2

Smash Version
13.0.1

ARCropolis Version
3.5.0

Can the crash be consistently reproduced?
Yes

At what moment does the crash happen?
The mod will cause crashes at random times only on the character select screen, when there is more than 2 players/CPUs/amiibo playing.

Details:
I was experiencing a similar issue prior to uploading to the latest version of ARCropolis. Once I updated, I troubleshooted each mod I have by selecting all the mods at once for each character in the game and no other mods on, and I produced a good majority of the time. I was also able to reproduce the same issue on the latest version by going one by one for each mod per character, still reproducing the issue. The crash only happens on the CSS, with more than 2 players/CPUs/amiibo playing. My only plugin is ARCropolis, with all mods in the directory "mods" in the ultimate folder on the root of my Fat32 SD card. The crash doesn't happen immediately, it usually will happen after I cycle through skins or browse the CSS, and takes a few seconds or close to a minute to happen.

Here are some logs I've taken from various attempts of reproducing the issue:

[01655147116_01006a800016e000.log](https://github.com/Raytwo/ARCropolis/files/8903962/01655147116_01006a800016e000.log)
[01655239636_01006a800016e000.log](https://github.com/Raytwo/ARCropolis/files/8903963/01655239636_01006a800016e000.log)
[01655239885_01006a800016e000.log](https://github.com/Raytwo/ARCropolis/files/8903964/01655239885_01006a800016e000.log)

Sorry for the trouble!

**Comment by tnaren97** [comment]
Can confirm, am also experiencing this issue on 3.5.0. Downgrading to 3.1.0 seems to fix it though. Maybe something added since then introduced a memory leak which is causing the crashes if too much is loaded in at once.

**Comment by arcenedon** [comment]
Update: Tried the latest version of ARCropolis, encountering the same issues

**Comment by NekoPrism** [comment]
Update: Issues still persist on the latest ARCropolis (3.6.2 as of writing)

**Comment by bakab0t** [comment]
encountering this issue even with no mods installed (on Arcropolis 3.6.2).
opening the color alt menu with 5+ player/cpu freezes and crashed the game.
downgrading acropolis to older versions didnt work for me.

**Reply by arcenedon** [reply]
same issue here.

**Comment by arcenedon** [comment]
can also confirm, this crash is reproduced on the latest version of arcropolis.
happened with 3 people

**Reply by dre154** [reply]
Make that 6, just confirmed with other modders that 3.6 and 3.7 are causing crashes when loading in more than 2+ players on a semi-consistent basis.

**Comment by dre154** [comment]
I can confirm that this is a 3.6+ issue. I've gone through my own mods as well as checked others and those who are reporting it all are saying the same thing; that loading in 2+ players will freeze/lock up/crash the game either immediately or some time AFTER having the icons hover on the screen for a bit. I've somehow managed to get 4 players to work for a time, but eventually the game will crash. Loading in 8 players seems to speed up the time before the crash as well. Notably, some crashes will stop the game and FREEZE it entirely...while after the first couple of crashes it will "lock up" the game where sound and music can be heard....but the cursor and screen will stay frozen. Since then I've tried downgrading to version 3.5.1 of Arcropolis but even this doesn't fix it...which is strange because prior to updating to 3.6.X+ I had absolutely zero issues with my mods whatsoever and nothing has changed since then even with one-sot effects active.

For the record and hopefully some assistance in ruling out variables: SDcard formatted to FAT32, latest version of atmosphere/hekate/arcropolis (3.7.2).

Plugins enabled: .DS_STORE, libarcropolis.nro, libbrawl_ness_effects.nro(Ness brawl effects plugin), libnn_hid_hook.nro, libnro_hook.nro, libparam_hook.nro, libsignpostgfx.nro (Sonic sign post victory screen), libsmashline_hook.nro, libone_slot_eff.nro (one-slot effects).

DISABLED PLUGINS: libno_music_reset.nro (No music reset in training mode or otherwise), libtraining_modpack_menu.nro, libtraining_modpack.nro

UPDATE: Upon further testing, at least on my end...trying the game with no mods whatsoever (holding L on boot) the game seems to load and play fine with up to 8 players (cpu) with no issues. Judging from previous threads as well as others users similar issues on previous versions of Arcropolis; my best guess is that loading in too many mods at once in the character selection screen seems to cause the game to freeze....HOW it causes it to freeze, I have no clue....could be too many high capacity files all being loaded too fast, too many big files altogether...I couldn't say for sure. All I know right now, is that it works fine for me without any mods enabled; but will consistently crash on 2+ players when mods are enabled even with low res (no 4k textures or big files etc etc) are selected. 

UPDATE 2: Now the game just seems to be on a time limit before crashing. Even with plugins disabled and no heavy-duty mods enabled, it just seems to freeze while the music still plays in the background. Not really sure what to make of it tbh.

CRASH REPORTS:
[01661914892_01006a800016e000.log](https://github.com/Raytwo/ARCropolis/files/9547495/01661914892_01006a800016e000.log)
[01662980453_01006a800016e000.log](https://github.com/Raytwo/ARCropolis/files/9547503/01662980453_01006a800016e000.log)
[01661921436_01006a800016e000.log](https://github.com/Raytwo/ARCropolis/files/9547510/01661921436_01006a800016e000.log)

**Reply by Raytwo** [reply]
I spot some smashline being the cause of crashes in there though.

**Comment by Raytwo** [comment]
Kinda very late on addressing this, but the version you're comparing the latest builds to doesn't have directory addition so that might be at play here.

Crash reports with bare minimum modpacks to reproduce them consistently are the best way to help us fix issues, so if any of you can provide that with as much context as possible, that'd be appreciated

**Reply by arcenedon** [reply]
Would you say that cosmetic mods like skins could possibly contribute to a crash? I have a preset on the latest firmware, where it's just skins.

**Reply by NekoPrism** [reply]
I have an old modpack that I have archived that I'm redoing because of this issue. Its around 14GB I think and only has skin mods. I can test it when I'm home and send the crash report here if something like that is what you're looking for.

Edit: I should also add that this issue, for me at least, has been consistently happening since ARC 2.0.0

**Comment by tnaren97** [comment]
Don't want to speak too soon but I think I might have a possible reason for the crashing. If you have the Ultimate Training Modpack installed, try deleting it in its entirety and reinstalling the latest version of Acropolis over it. After doing this, I had no problems loading up my 15 Gb modpack and sitting on the CSS with 8 characters selected cycling through alts for at least 5 solid minutes. Could be I just got lucky but it's worth testing.

**Reply by mRama3415** [reply]
still not having problems after doing this? has this worked for anyone else?

**Comment by Raytwo** [comment, marked as answer]
Fixed with ARCropolis 3.8.0
